### ISSUE TITLE
npm install error on windows 10 

### ISSUE THREAD

**Issue by higherstar** [nodejs/node-gyp#1788]
Hello Everybody
I installed node 8.11.2, installed windows-build-tools, also installed Visual Studio Community 2017 including C++ Build Toos, Windows 10 SDK as well.
And then tried to install npm i.
But I am still getting these errors.

--------------------------------------------

```
C:\Data\CropSpec\node_modules\bufferutil>if not defined npm_config_node_gyp (node "C:\Users\seed\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Users\seed\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  bufferutil.cc
  win_delay_load_hook.cc
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan_new.h(34): warning C4996: 'v8::Value::ToInt32': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(9796): note: see declaration of 'v8::Value::ToInt32' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan_new.h(39): warning C4996: 'v8::Value::ToUint32': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(9790): note: see declaration of 'v8::Value::ToUint32' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan_implementation_12_inl.h(49): warning C4996: 'v8::BooleanObject::New': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(4845): note: see declaration of 'v8::BooleanObject::New' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan_implementation_12_inl.h(166): warning C4996: 'v8::ScriptCompiler::Compile': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(1423): note: see declaration of 'v8::ScriptCompiler::Compile' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan_implementation_12_inl.h(173): warning C4996: 'v8::ScriptCompiler::Compile': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(1423): note: see declaration of 'v8::ScriptCompiler::Compile' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan_implementation_12_inl.h(205): warning C4996: 'v8::String::NewFromOneByte': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(2679): note: see declaration of 'v8::String::NewFromOneByte' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan_implementation_12_inl.h(217): warning C4996: 'v8::String::NewExternal': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(2718): note: see declaration of 'v8::String::NewExternal' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan_implementation_12_inl.h(237): warning C4996: 'v8::ScriptCompiler::CompileUnbound': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(1404): note: see declaration of 'v8::ScriptCompiler::CompileUnbound' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan_implementation_12_inl.h(244): warning C4996: 'v8::ScriptCompiler::CompileUnbound': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(1404): note: see declaration of 'v8::ScriptCompiler::CompileUnbound' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(263): error C2995: 'v8::Local<T> _NanEnsureLocal(v8::Local<T>)': function template has already been defined (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(256): note: see declaration of '_NanEnsureLocal' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(289): warning C4996: 'v8::Isolate::IdleNotification': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(7481): note: see declaration of 'v8::Isolate::IdleNotification' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(473): error C2039: 'GCEpilogueCallback': is not a member of 'v8::Isolate' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(6732): note: see declaration of 'v8::Isolate' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(473): error C2065: 'GCEpilogueCallback': undeclared identifier (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(474): error C2146: syntax error: missing ')' before identifier 'callback' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(474): error C2143: syntax error: missing ';' before '{' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(474): error C2447: '{': missing function header (old-style formal list?) (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(479): error C2039: 'GCEpilogueCallback': is not a member of 'v8::Isolate' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(6732): note: see declaration of 'v8::Isolate' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(479): error C2065: 'GCEpilogueCallback': undeclared identifier (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(479): error C2146: syntax error: missing ')' before identifier 'callback' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(479): error C2143: syntax error: missing ';' before '{' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(479): error C2447: '{': missing function header (old-style formal list?) (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(484): error C2039: 'GCPrologueCallback': is not a member of 'v8::Isolate' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(6732): note: see declaration of 'v8::Isolate' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(484): error C2065: 'GCPrologueCallback': undeclared identifier (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(485): error C2146: syntax error: missing ')' before identifier 'callback' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(485): error C2143: syntax error: missing ';' before '{' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(485): error C2447: '{': missing function header (old-style formal list?) (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(490): error C2039: 'GCPrologueCallback': is not a member of 'v8::Isolate' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(6732): note: see declaration of 'v8::Isolate' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(490): error C2065: 'GCPrologueCallback': undeclared identifier (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(490): error C2146: syntax error: missing ')' before identifier 'callback' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(490): error C2143: syntax error: missing ';' before '{' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(490): error C2447: '{': missing function header (old-style formal list?) (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(569): error C2039: 'WeakCallbackData': is not a member of 'v8' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(64): note: see declaration of 'v8' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(569): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(569): error C2988: unrecognizable template declaration/definition (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(569): error C2143: syntax error: missing ',' before '<' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(661): error C2039: 'smalloc': is not a member of 'node' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\node_object_wrap.h(29): note: see declaration of 'node' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(661): error C3083: 'smalloc': the symbol to the left of a '::' must be a type (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(661): error C2039: 'FreeCallback': is not a member of 'node' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\node_object_wrap.h(29): note: see declaration of 'node' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(661): error C2061: syntax error: identifier 'FreeCallback' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(665): error C2065: 'callback': undeclared identifier (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(665): error C2065: 'hint': undeclared identifier (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(672): error C2665: 'node::Buffer::New': none of the 4 overloads could convert all the argument types (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\node_buffer.h(67): note: could be 'v8::MaybeLocal<v8::Object> node::Buffer::New(v8::Isolate *,char *,size_t)' (compiling source file ..\src\bufferutil.cc)
  c:\users\seed\.node-gyp\8.11.2\include\node\node_buffer.h(55): note: or       'v8::MaybeLocal<v8::Object> node::Buffer::New(v8::Isolate *,v8::Local<v8::String>,node::encoding)' (compiling source file ..\src\bufferutil.cc)
  c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(672): note: while trying to match the argument list '(v8::Isolate *, const char *, uint32_t)' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(676): error C2440: 'return': cannot convert from 'v8::MaybeLocal<v8::Object>' to 'v8::Local<v8::Object>' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(676): note: No constructor could take the source type, or constructor overload resolution was ambiguous (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(683): error C2039: 'Use': is not a member of 'node::Buffer' (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\node_buffer.h(32): note: see declaration of 'node::Buffer' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(683): error C3861: 'Use': identifier not found (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(710): warning C4996: 'v8::ScriptCompiler::Compile': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(1423): note: see declaration of 'v8::ScriptCompiler::Compile' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(717): warning C4996: 'v8::ScriptCompiler::Compile': was declared deprecated (compiling source file ..\src\bufferutil.cc) [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\v8.h(1423): note: see declaration of 'v8::ScriptCompiler::Compile' (compiling source file ..\src\bufferutil.cc)
c:\data\cropspec\node_modules\bufferutil\src\bufferutil.cc(30): error C2668: 'node::NODE_SET_METHOD': ambiguous call to overloaded function [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\node.h(290): note: could be 'void node::NODE_SET_METHOD(v8::Local<v8::Object>,const char *,v8::FunctionCallback)' (compiling source file ..\src\bufferutil.cc)
  c:\users\seed\.node-gyp\8.11.2\include\node\node.h(277): note: or       'void node::NODE_SET_METHOD(v8::Local<v8::Template>,const char *,v8::FunctionCallback)' (compiling source file ..\src\bufferutil.cc)
  c:\data\cropspec\node_modules\bufferutil\src\bufferutil.cc(30): note: while trying to match the argument list '(v8::Local<v8::FunctionTemplate>, const char [7], overloaded-function)'
c:\data\cropspec\node_modules\bufferutil\src\bufferutil.cc(31): error C2668: 'node::NODE_SET_METHOD': ambiguous call to overloaded function [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\node.h(290): note: could be 'void node::NODE_SET_METHOD(v8::Local<v8::Object>,const char *,v8::FunctionCallback)' (compiling source file ..\src\bufferutil.cc)
  c:\users\seed\.node-gyp\8.11.2\include\node\node.h(277): note: or       'void node::NODE_SET_METHOD(v8::Local<v8::Template>,const char *,v8::FunctionCallback)' (compiling source file ..\src\bufferutil.cc)
  c:\data\cropspec\node_modules\bufferutil\src\bufferutil.cc(31): note: while trying to match the argument list '(v8::Local<v8::FunctionTemplate>, const char [5], overloaded-function)'
c:\data\cropspec\node_modules\bufferutil\src\bufferutil.cc(32): error C2668: 'node::NODE_SET_METHOD': ambiguous call to overloaded function [C:\Data\CropSpec\node_modules\bufferutil\build\bufferutil.vcxproj]
  c:\users\seed\.node-gyp\8.11.2\include\node\node.h(290): note: could be 'void node::NODE_SET_METHOD(v8::Local<v8::Object>,const char *,v8::FunctionCallback)' (compiling source file ..\src\bufferutil.cc)
  c:\users\seed\.node-gyp\8.11.2\include\node\node.h(277): note: or       'void node::NODE_SET_METHOD(v8::Local<v8::Template>,const char *,v8::FunctionCallback)' (compiling source file ..\src\bufferutil.cc)
  c:\data\cropspec\node_modules\bufferutil\src\bufferutil.cc(32): note: while trying to match the argument list '(v8::Local<v8::FunctionTemplate>, const char [6], overloaded-function)'
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\seed\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\node-gyp\lib\build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\seed\\AppData\\Roaming\\nvm\\v8.11.2\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Data\CropSpec\node_modules\bufferutil
gyp ERR! node -v v8.11.2
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
```
-----------------------
Please help me!

**Comment by rvagg** [comment]
The errors you're getting say that the version of "bufferutil" you're compiling isn't compatible with the version of Node you're using. You must be using quite an old version of it though because it's been updated. Fetch a newer version and try again and those errors should go away. If bufferutil is pinned as a dependency of something else (CropSpec?) then you will need to update the version there, or if you're not the maintainer, get the maintainer to update it.

**Comment by BrutalBirdie** [comment]
Just to add this if someone stumbles over it.
```
C:\WINDOWS\system32>npm install -g node-lua
npm WARN npm npm does not support Node.js v10.0.0
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 4, 6, 7, 8, 9.
npm WARN npm You can find the latest version at https://nodejs.org/
npm WARN npm npm does not support Node.js v10.0.0
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 4, 6, 7, 8, 9.
npm WARN npm You can find the latest version at https://nodejs.org/

> node-lua@1.0.4 install C:\Program Files\nodejs\node_modules\node-lua
> node-gyp rebuild


C:\Program Files\nodejs\node_modules\node-lua>if not defined npm_config_node_gyp (node "C:\Users\eha-tower\AppData\Roaming\nvm\v10.0.0\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Users\eha-tower\AppData\Roaming\nvm\v10.0.0\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Die Projekte in dieser Projektmappe werden nacheinander erstellt. Um eine parallele Erstellung zu ermöglichen, müssen Sie den Schalter "/m" hinzufügen.
  move_lua
  Der Befehl "cp" ist entweder falsch geschrieben oder
  konnte nicht gefunden werden.
C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(209,5): error MSB6006: "cmd.exe" wurde mit dem Code 1 beendet. [C:\Program Files\nodejs\node_modules\node-lua\build\nodelua.
vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\eha-tower\AppData\Roaming\nvm\v10.0.0\node_modules\npm\node_modules\node-gyp\lib\build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:225:12)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\eha-tower\\AppData\\Roaming\\nvm\\v10.0.0\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Program Files\nodejs\node_modules\node-lua
gyp ERR! node -v v10.0.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-lua@1.0.4 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-lua@1.0.4 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\eha-tower\AppData\Roaming\npm-cache\_logs\2019-06-20T10_31_17_625Z-debug.log
```
Command cp not found, "Uhhmm I am on windows?"
Tried with the git-bash

```
$ npm install -g node-lua
npm WARN npm npm does not support Node.js v10.0.0
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 4, 6, 7, 8,                                                                                                                                                                                     9.
npm WARN npm You can find the latest version at https://nodejs.org/
npm WARN npm npm does not support Node.js v10.0.0
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 4, 6, 7, 8,                                                                                                                                                                                     9.
npm WARN npm You can find the latest version at https://nodejs.org/

> node-lua@1.0.4 install C:\Program Files\nodejs\node_modules\node-lua
> node-gyp rebuild


C:\Program Files\nodejs\node_modules\node-lua>if not defined npm_config_node_gyp                                                                                                                                                                                     (node "C:\Users\eha-tower\AppData\Roaming\nvm\v10.0.0\node_modules\npm\node_mod                                                                                                                                                                                    ules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" re                                                                                                                                                                                    build )  else (node "C:\Users\eha-tower\AppData\Roaming\nvm\v10.0.0\node_modules                                                                                                                                                                                    \npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Die Projekte in dieser Projektmappe werden nacheinander erstellt. Um eine parall                                                                                                                                                                                    ele Erstellung zu erm▒glichen, m▒ssen Sie den Schalter "/m" hinzuf▒gen.
  move_lua
  utils.cc
  luastate.cc
  nodelua.cc
  win_delay_load_hook.cc
c:\program files\nodejs\node_modules\node-lua\src\utils.cc(11): warning C4996: 'v8::String::Utf8Value::Utf8Value': wurde als veraltet deklariert [C:\Program Files\nodejs\node_modules\node-lua\build\nodelua.vcxproj]
  c:\users\eha-tower\.node-gyp\10.0.0\include\node\v8.h(2821): note: Siehe Deklaration von "v8::String::Utf8Value::Utf8Value"
c:\program files\nodejs\node_modules\node-lua\src\luastate.cc(115): warning C4996: 'Nan::MakeCallback': wurde als veraltet deklariert [C:\Program Files\nodejs\node_modules\node-lua\build\nodelua.vcxproj]
  c:\users\eha-tower\appdata\roaming\nvm\v10.0.0\node_modules\node-lua\node_modules\nan\nan.h(1024): note: Siehe Deklaration von "Nan::MakeCallback"
     Bibliothek "C:\Program Files\nodejs\node_modules\node-lua\build\Release\nodelua.lib" und Objekt "C:\Program Files\nodejs\node_modules\node-lua\build\Release\nodelua.exp" werden erstellt.
  Code wird generiert.
  All 452 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Codegenerierung ist abgeschlossen.
  nodelua.vcxproj -> C:\Program Files\nodejs\node_modules\node-lua\build\Release\\nodelua.node
+ node-lua@1.0.4
added 2 packages in 6.272s
```

**Comment by higherstar** [comment]
@rvagg so you mean I need to upgrade the npm version which I am using to install dependencies?

**Comment by rvagg** [comment]
@higherstar no, the version of "bufferutil" you are trying to build isn't compatible, you need to update it, wherever it's being pulled in.

```
C:\Data\CropSpec\node_modules\bufferutil>
```

Something in your project, maybe your project itself, is using "bufferutil", a package from npm, it's too old for your version of Node. The compile is failing because the C++ in the version of bufferutil you have isn't compatible with the version of Node you are using.

Use `npm ls` to figure out where "bufferutil" is coming from. The latest version is 4.0.1, that's what you should be aiming to include: https://github.com/websockets/bufferutil, if it's being depended upon by some other dependency of your project, you should hunt them down and ask them to upgrade.

`c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(263): error C2995: 'v8::Local<T> _NanEnsureLocal(v8::Local<T>)': function template has already been defined`

`c:\data\cropspec\node_modules\bufferutil\node_modules\nan\nan.h(473): error C2039: 'GCEpilogueCallback': is not a member of 'v8::Isolate'`

These kinds of errors are the indication of the mismatch. "nan.h" is a compatibility layer between addon C++ code and Node + V8, but it's too old for your version of Node.

The alternative is to downgrade your version of Node. You're using 8.11.2 which is already pretty old, so the version of bufferutil you're using must be quite old. You could try going back to Node 6, but that's not really recommended because it's not supported anymore.

Basically, upgrade your dependencies, chase down authors of dependencies that are using old versions of addons that aren't compiling.
